dts/bindings/gpio/litex,gpio.yaml

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright (c) 2019 Antmicro <www.antmicro.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: Litex GPIO
+
+description: >
+    This is a representation of the Litex GPIO nodes
+
+inherits:
+    !include base.yaml
+
+properties:
+    compatible:
+      category: required
+      constraint: "litex,gpio"
+
+    direction:
+      type: string
+      category: required
+      description: Direction of the port "in" or "out"
+
+    reg:
+      category: required
+
+    label:
+      category: required
+
+    pin-amount:
```

Comment:
Maybe `nr-gpios` instead.

Reply:
It looks like properties with `gpios` in the name are handled in a special way by the DTS extractor script - see [here](https://github.com/zephyrproject-rtos/zephyr/blob/303ef27535cc7df590d76584bd0c0ee991d222fd/scripts/dts/extract_dts_includes.py#L87).

As a result after the suggested rename `sanitycheck` fails with: 
```
device tree error: bad phandle in <Property 'nr-gpios' at '/soc/gpio@e0003000' in 'litex_vexriscv.dts.pre.tmp'>
```

Reply:
I think this is because the type of property is not correct.  It should not be 'array' but 'int'

Reply:
Changing the type to `int` did not help.

I noticed that writing `nr-gpios: <&intc0>` in DTS solves the problem of "bad phandle", but (apart from not having any sense in this context) it creates another one:

```
device tree error: missing data after phandle in <Property 'nr-gpios' at '/soc/gpio@e0003000' in 'litex_vexriscv.dts.pre.tmp'>
```

Reverting type to `array` and writing `nr-gpios <&intc0 1 1>` solves parsing problem, but then `gen_defines.py` script fails in `write_gpio` function.

I have a strong feeling that properties with "gpios" in the name are expected to reference interrupt controller and are used in the interrupts context.

This is not what we intend for LiteX GPIO port.

Reply:
Can you push the PR with the change to it being 'int' and that shows the CI error.  It will be easier for me to take a look to see what's going on than.

Reply:
@galak Pushed.